### PR TITLE
Feat/exclude unhealthy gpu devices

### PR DESCRIPTION
**Pull request by WingkaiHo**
Exclude unhealhy gpu device by list of node annotations `volcano.sh/gpu-unhealthy-ids`, which patch by volcano device plugin

https://github.com/volcano-sh/devices/pull/24

### CODE REVIEW

File: pkg/scheduler/api/node_info.go [hunk 1]
```diff
@@ -327,6 +328,11 @@ func (ni *NodeInfo) setNodeGPUInfo(node *v1.Node) {
 	for i := 0; i < int(gpuNumber); i++ {
 		ni.GPUDevices[i] = NewGPUDevice(i, memoryPerCard)
 	}
+	unhealthyGPUs := ni.getUnhealthyGPUs(node)
+	for id := range unhealthyGPUs {
+		klog.Info("delete unhealthy gpu id %d from GPUDevices", id)
```
Comment (Thor-wl) [inline pkg/scheduler/api/node_info.go:333]: Please set a reasonable log level.

File: pkg/scheduler/api/node_info.go [hunk 2]
```diff
@@ -580,3 +586,24 @@ func (ni *NodeInfo) SubGPUResource(pod *v1.Pod) {
 		}
 	}
 }
+
+// getUnhealthyGPUs returns all the unhealthy GPU id.
+func (ni *NodeInfo) getUnhealthyGPUs(node *v1.Node) (unhealthyGPUs map[int]bool) {
```
Comment (Thor-wl) [inline pkg/scheduler/api/node_info.go:591]: Why returns a map whose value is bool? According to the following logic, although the current data struct also works, it seems a simpler data struct meets the requirement such as array, which may be more memory friendly.

File: pkg/scheduler/api/node_info.go [hunk 3]
```diff
@@ -580,3 +586,24 @@ func (ni *NodeInfo) SubGPUResource(pod *v1.Pod) {
...
+	unhealthyGPUs = map[int]bool{}
+	devicesStr, ok := node.Annotations[UnhealthyGPUIDs]
```
Comment (Thor-wl) [inline pkg/scheduler/api/node_info.go:593]: BTW, @william-wang @shinytang6 Does device plugin report the unhealthy GPU ids? And these ids will be patched to node object as an annotation?

File: pkg/scheduler/api/node_info.go (hunk 1 above)
Comment (WingkaiHo) [reply pkg/scheduler/api/node_info.go]: ok

File: pkg/scheduler/api/node_info.go (hunk 2 above)
Comment (WingkaiHo) [reply pkg/scheduler/api/node_info.go]: ok, fix

**Review by Thor-wl — COMMENTED** [review]
Combinded with [PR-24](https://github.com/volcano-sh/devices/pull/24), it's generally OK to me now. Thanks for your contributioin!
/lgtm

File: pkg/scheduler/api/node_info.go (hunk 3 above)
Comment (william-wang) [reply pkg/scheduler/api/node_info.go:593]: The device plugin patch the unhealthy gpu id.  We need to make clear the following pr in device plugin firstly and then work on this one.
https://github.com/volcano-sh/devices/pull/24

Comment (WingkaiHo) [reply pkg/scheduler/api/node_info.go:593]: @william-wang volcano-sh/devices#24 have been merge。 Anyother questions in this pr.

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**